### PR TITLE
Make self-hosted data migration easier

### PR DESCRIPTION
**Pull request by ruslandoga**
### Changes

This PR makes self-hosted migration a bit easier.

- Defaulting to v2 allows users to skip setting `V2_MIGRATION_DONE` env var. It also forces them to migrate since until they do, dashboard wouldn't show pre-v2-startup data.
- Allowing `N` prompt answer defaults and custom prompt messages provides more info to the users when they are dropping `events_v2` and `sessions_v2` tables. If they are running migrations some time after starting the new Plausible version, they'd have some data in those tables (due to the change above) and we are telling them exactly how much and defaulting to N for the answer. I'm keeping the ability to drop v2 tables in case someone botches the migration on first attempt and wants to try again (in which case they do want to drop v2 tables).

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] This change does not need a documentation update

### Dark mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/data_migration/numeric_ids.ex [hunk 1]
```diff
@@ -90,8 +89,28 @@ defmodule Plausible.DataMigration.NumericIDs do
...
 
-    {:ok, _} = run_sql_fn.("drop-events-v2", cluster?: cluster?)
-    {:ok, _} = run_sql_fn.("drop-sessions-v2", cluster?: cluster?)
+    drop_v2_extra_opts = fn table ->
+      case @repo.query("select count(*) from {$0:Identifier}", [table]) do
+        {:ok, %{rows: [[count]]}} when count > 0 ->
+          [
+            prompt_message: "The table contains #{count} rows. Execute?",
+            prompt_default_choice: :no
+          ]
```
Comment (ruslandoga) [inline lib/plausible/data_migration/numeric_ids.ex:98]: Screenshot 2023-04-21 at 10 47 45

**Comment by ruslandoga** [comment]
I also made [a PR that removes v1](https://github.com/ruslandoga/analytics/pull/90) but thought that's it's too much and possibly incomplete or worse, contains errors.

File: priv/data_migrations/NumericIDs/sql/insert-into-tmp-events-v2.sql.eex [hunk 2]
```diff
@@ -10,7 +10,7 @@ INSERT INTO tmp_events_v2 (
   transferred_from
 )
 SELECT
-  (SELECT site_id from domains_lookup WHERE domain = domain LIMIT 1),
```
Comment (ruslandoga) [inline priv/data_migrations/NumericIDs/sql/insert-into-tmp-events-v2.sql.eex:13]: Subqueries in ClickHouse don't seem to have access to parent queries so this subselect didn't work as expected: it always returned the first site id.

File: config/.env.dev [hunk 3]
```diff
@@ -17,4 +17,3 @@ GOOGLE_CLIENT_SECRET=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
 
 PROMEX_DISABLED=false
 SITE_DEFAULT_INGEST_THRESHOLD=1000000
-V2_MIGRATION_DONE=1
```
Comment (aerosol) [inline config/.env.dev:20]: Maybe then we can also remove the additional CI test run that's running for V1 - if you look at the GitHub workflow for PRs. I'm not entirely sure if that's not too early, but probably not.

File: lib/plausible/data_migration.ex [hunk 4]
```diff
@@ -12,20 +12,43 @@ defmodule Plausible.DataMigration do
       @dir dir
       @repo repo
 
-      def run_sql_confirm(name, assigns \\ []) do
-        query = unwrap_with_io(name, assigns)
+      def run_sql_confirm(name, options \\ []) do
```
Comment (aerosol) [inline lib/plausible/data_migration.ex:15]: :raised_hands:

File: priv/data_migrations/NumericIDs/sql/insert-into-tmp-sessions-v2.sql.eex [hunk 5]
```diff
@@ -6,12 +6,13 @@ operating_system_version, subdivision1_code, subdivision2_code, city_geoname_id,
...
-  (SELECT site_id from domains_lookup WHERE domain = domain LIMIT 1),
+  site_id,
   session_id, sign, user_id, hostname, timestamp, start, is_bounce, entry_page, 
   exit_page, pageviews, events, duration, referrer, referrer_source, country_code, 
   screen_size, operating_system, browser, utm_medium, utm_source, utm_campaign, browser_version, 
   operating_system_version, subdivision1_code, subdivision2_code, city_geoname_id, utm_content, 
   utm_term, transferred_from, entry_meta.key, entry_meta.value
 FROM
   sessions
+INNER JOIN domains_lookup ON domains_lookup.domain = sessions.domain
```
Comment (aerosol) [inline priv/data_migrations/NumericIDs/sql/insert-into-tmp-sessions-v2.sql.eex:17]: Great, that should work for smaller datasets!

**Review by aerosol — APPROVED** [review]

File: priv/data_migrations/NumericIDs/sql/insert-into-tmp-sessions-v2.sql.eex (hunk 5 above)
Comment (ruslandoga) [reply priv/data_migrations/NumericIDs/sql/insert-into-tmp-sessions-v2.sql.eex:17]: Ah, I was wondering why you chose a different method. Is there a performance cost to a join like this?

Comment (aerosol) [reply priv/data_migrations/NumericIDs/sql/insert-into-tmp-sessions-v2.sql.eex:17]: I think there is a notion of joins being discouraged if you can avoid them. I didn't want to risk it in the initial implementation.

File: priv/data_migrations/NumericIDs/sql/insert-into-tmp-events-v2.sql.eex (hunk 2 above)
Comment (aerosol) [reply priv/data_migrations/NumericIDs/sql/insert-into-tmp-events-v2.sql.eex:13]: Oh I see. I freaked out a little, but then I remembered we used a dict lookup in the first implementation.

File: config/.env.dev (hunk 3 above)
Comment (aerosol) [reply config/.env.dev:20]: Great!

File: priv/data_migrations/NumericIDs/sql/insert-into-tmp-events-v2.sql.eex (hunk 2 above)
Comment (ruslandoga) [reply priv/data_migrations/NumericIDs/sql/insert-into-tmp-events-v2.sql.eex:13]: I ran the dict version and it worked great :) The only problem was setting the dict up...

**Pull request merged**